src/test/java/org/broadinstitute/barclay/argparser/TaggedArgumentTest.java

```diff
@@ -0,0 +1,301 @@
+        }
+
+        @Override
+        public void setTag(final String tagName) {
+            this.tagName = tagName;
+        }
+
+        @Override
+        public String getTag() {
+            return tagName;
+        }
+
+        @Override
+        public void setTagAttributes(final Map<String, String> attributes) {
+            this.tagAttributes.putAll(attributes);
+        }
+
+        @Override
+        public Map<String, String> getTagAttributes() {
+            return tagAttributes;
+        }
+
+        @Override
+        public String toString() { return argValue; }
+    }
+
+    @CommandLineProgramProperties(
+            summary = "Test tagged aruments",
+            oneLineSummary = "Test tagged aruments",
+            programGroup = TestProgramGroup.class
+    )
+    public class TaggableArguments {
+
+        @PositionalArguments(minElements = 0, maxElements=2)
+        public List<String> positionalArgs = new ArrayList<>();
+
+        @Argument(shortName="t", fullName="tFullName", doc="Tag target arg", optional = true)
+        public List<TaggableArg> taggableArgList = new ArrayList<>();
+
+        @Argument(shortName="s", fullName="scalarArg", optional = true)
+        public Integer taggableInteger = 17;
+
+        @Argument(shortName="n", fullName="notTaggable", optional = true)
+        public String notTaggable;
+    }
+
+    // Non-generic key-value class for easy addition to statically initialized arrays in data provider
+    private static class KeyValuePair{
+        private final String key, value;
+
+        public static KeyValuePair of(final String key, final String right) { return new KeyValuePair(key, right); }
+
+        private KeyValuePair(final String left, final String right) { this.key = left; this.value = right; }
+        public String getKey() { return this.key; }
+        public String getValue() { return this.value; }
+    }
+
+    @SuppressWarnings("unchecked")
+    @DataProvider(name = "GoodTaggedArguments")
+    public Object[][] goodTaggedArguments() {
+        return new Object[][][]{
+                //TODO: file args, tagged scalar/enum (?), intermix tagged with positional args
+                {
+                        // single value in a collection, no attributes, short name
+                        new String[]{"--t:tumor", "tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "tumor.bam")},
+                        new KeyValuePair[][]{{}}
+                },
+                {
+                        // single value in a collection, no attributes, short name
+                        new String[]{"--t:tumor", "/tumor.bam"},
+                        new KeyValuePair[]{KeyValuePair.of("tumor", "/tumor.bam")},
+                        new KeyValuePair[][]{{}}
+                },
+                {
+                        // single tagged value in a collection, no attributes, short name
+                        new String[]{"--t:tumor", "gcs://my/tumor.bam"},
```

Comment:
Add additional test cases: `"--t:tumor", "hostName:1234/tumor.bam"` and `"--t:tumor", "http://hostName:1234/tumor.bam"`